server/etcdserver/api/snap/snapshotter.go

```diff
@@ -88,7 +89,9 @@ func (s *Snapshotter) save(snapshot *raftpb.Snapshot) error {
 	spath := filepath.Join(s.dir, fname)
 
 	fsyncStart := time.Now()
+	cancel := watchdog.Register("save v2 snapshot")
```

Comment:
Is the watchdog global.   I was assuming watchdog is an instance... and callers of watchdog needs to have reference to the object... rather being a package.

Reply:
Yes, it's global, but it's only accessed via exported functions.

It will be called by multiple different places (e.g. wal, batchTx, etc.), it seems not good and incontinent to assign the watchdog instance to each of the objects (e.g.  [WAL](https://github.com/etcd-io/etcd/blob/c6d8b65ea7a77556f44a826eecf08c83cc966bca/server/storage/wal/wal.go#L72), [batchTx](https://github.com/etcd-io/etcd/blob/c6d8b65ea7a77556f44a826eecf08c83cc966bca/server/storage/backend/batch_tx.go#L61) and [Snapshotter](https://github.com/etcd-io/etcd/blob/c6d8b65ea7a77556f44a826eecf08c83cc966bca/server/etcdserver/api/snap/snapshotter.go#L53) )